mlabs/src/Mlabs/NFT/Validation.hs

```diff
@@ -246,379 +246,383 @@ mintPolicy appInstance =
-          withAuctionState $ \auctionState ->
-            case as'highestBid auctionState of
-              Nothing -> True
-              Just (AuctionBid bid _bidder) ->
-                correctPaymentCheck bid
-
-        auctionCorrectPaymentOwner :: Bool
-        auctionCorrectPaymentOwner = auctionCorrectPayment correctPaymentOwner
-
-        auctionCorrectPaymentAuthor :: Bool
-        auctionCorrectPaymentAuthor = auctionCorrectPayment correctPaymentAuthor
-
-        auctionCorrectPaymentOnlyAuthor :: Bool
-        auctionCorrectPaymentOnlyAuthor =
-          withAuctionState $ \auctionState ->
-            case as'highestBid auctionState of
-              Nothing -> True
-              Just (AuctionBid bid _) ->
-                correctPaymentOnlyAuthor bid
-
-        correctBidRefund :: Bool
-        correctBidRefund =
-          withAuctionState $ \auctionState ->
-            case as'highestBid auctionState of
-              Nothing -> True
-              Just (AuctionBid bid bidder) ->
-                valuePaidTo info (getUserId bidder) == Ada.lovelaceValueOf bid
-
-        correctInputValue :: Bool
-        correctInputValue =
-          case findOwnInput ctx of
-            Nothing -> traceError "findOwnInput: Nothing"
-            Just (TxInInfo _ out) ->
-              case mauctionState of
-                Nothing -> traceError "mauctionState: Nothing"
-                Just as -> case as'highestBid as of
-                  Nothing -> tokenValue == txOutValue out
-                  Just hb -> txOutValue out == (tokenValue <> Ada.lovelaceValueOf (ab'bid hb))
-
-        auctionBidValueSupplied :: Integer -> Bool
-        auctionBidValueSupplied redeemerBid =
-          case fmap snd . getOutputDatumsWithTx @DatumNft $ ctx of
-            [out] -> txOutValue out == tokenValue <> Ada.lovelaceValueOf redeemerBid
-            [] -> traceError "auctionBidValueSupplied: expected exactly one continuing output, got none"
-            _ -> traceError "auctionBidValueSupplied: expected exactly one continuing output, got several instead"
-
-        auctionCorrectNewOwner :: Bool
-        auctionCorrectNewOwner =
-          withAuctionState $ \auctionState ->
-            case as'highestBid auctionState of
-              Nothing -> True
-              Just (AuctionBid _ bidder) ->
-                bidder == newOwner
-          where
-            newOwner = info'owner newNodeInfo
-
-        auctionConsistentCloseDatum :: Bool
-        auctionConsistentCloseDatum =
-          -- Checking that all fields remain the same except owner
-          info'id newNodeInfo == info'id nInfo
-            && info'share newNodeInfo == info'share nInfo
-            && info'author newNodeInfo == info'author nInfo
-            && info'price newNodeInfo == info'price nInfo
-            && checkOwner
-          where
-            checkOwner = withAuctionState $ \auctionState ->
+  traceIfFalse "Fees are not Paid Correctly." True -- todo
```

Comment:
Are you sure about this @cstml ? You seem to have used a much older file here

Reply:
Cheers for pointing out, you are right, it was rebased on top of an old `staging`. I've amended.